io-engine/src/lvs/lvs_store.rs

```diff
@@ -731,6 +783,35 @@ impl Lvs {
         Ok(())
     }
 
+    /// Grows the online (live) pool.
+    #[tracing::instrument(level = "debug", err)]
+    pub async fn grow(&self) -> Result<(), LvsError> {
+        info!("{self:?}: growing lvs...");
+
+        let (s, r) = pair::<i32>();
+
+        unsafe {
+            let lvs = self.as_inner_ptr();
+
+            // Update block count on spdk_bs_bdev.
+            spdk_bdev_update_bs_blockcnt((*lvs).bs_dev);
```

Comment:
Is this update automatically reverted in case grow fails?

Reply:
No, but it is not needed.

In SPDK blobstore, there's another layer of abstraction/indirection on top of SPDK bdev: spdk_bs_bdev. It is needed for provide a uniform interface for possible Blob storage backends: normal bdev, a virtual zero-filled backend for thin provisioning, and some other stuff. Basically, it is like a simplified version of bdev struct, added for needs of blob store code.

For performance reasons (my guess), spdk_bs_bdev struct caches the underlying device size. This function spdk_bdev_update_bs_blockcnt(), it just updates this cached device size so it reflects the actual bdev size. It has nothing to do with pool grow itself.

(My opinion is that blob store should update that cache size automatically in response to bdev resize event, but SPDK does not implement that. Probably, they will fix that in future.)